### PR TITLE
aws: improve error.message and add event.kind:pipeline_error for pipeline errors

### LINKED ISSUE #11090: [aws]: pipeline_error "Cannot invoke \"java.lang.CharSequence.length()\" because \"this.wrapped\" is null"
### Integration Name

AWS [aws]

### Dataset Name

aws.guardduty aws.elb_logs 

### Integration Version

2.24.1

### Agent Version

8.15.0

### Agent Output Type

elasticsearch

### Elasticsearch Version

8.15.0

### OS Version and Architecture

ubuntu 20.04 LTS

### Software/API Version

AWS

### Error Message

freshly started ingesting AWS logs, for now Guardduty and ELB logs. all of them end up with event.kind: pipeline_error and error.message: "Cannot invoke \"java.lang.CharSequence.length()\" because \"this.wrapped\" is null" 

### Event Original

[7b7f2b74-4804-3097-9879-07bb6e432af9.jsonl.json](https://github.com/user-attachments/files/16960528/7b7f2b74-4804-3097-9879-07bb6e432af9.jsonl.json)
For Guardduty, created some example findings, as can be found above.

resulting document:
[resulting_document.json](https://github.com/user-attachments/files/16960615/resulting_document.json)

### What did you do?

ingest AWS guardduty and ELB logs from S3 via SQS notifications

### What did you see?

pipeline_error and error.message

### What did you expect to see?

no pipeline_error and no error.message

### Anything else?

at a first glance, the fields seem to be parsed all well, dashboards show data. the pipeline_error and error.message are very irritating.

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

Also fix instances of incorrect yaml for script processors.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #11090
- Requires #11397

### CODE REVIEW

**Review by kcreddy — APPROVED** [review]

**Review by ishleenk17 — APPROVED** [review]

File: packages/aws/data_stream/apigateway_logs/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -89,5 +89,8 @@ processors:
       ignore_failure: true
 on_failure:
   - set:
-      field: 'error.message'
-      value: '{{ _ingest.on_failure_message }}'
+      field: event.kind
+      value: pipeline_error
+  - append:
+      field: error.message
+      value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
```
Comment (andrewkroh) [inline packages/aws/data_stream/apigateway_logs/elasticsearch/ingest_pipeline/default.yml:96]: If you have an easy way to update these, consider making the "with tag" part conditional on whether the processor has a tag.

For example:

https://github.com/andrewkroh/go-examples/blob/ee88091db24ef39d92d4431fa4a112e6fcb00d02/ecs-update/ecs-update.go#L1697-L1700

**Review by andrewkroh — APPROVED** [review]

**Comment by efd6** [comment]
@romulets I would like to understand how to resolve the conflicts in this PR. There are three possible approaches (four):
1. Merge this as 2.27.0-preview01 and leave the benefits unavailable to users until the the features in 2.26.0-preview01 are accepted as GA.
2. Merge this as 2.26.0 after reverting the changes in 2.26.0-preview01.
3. Merge this as 2.27.0 and release the features in 2.26.0-preview01.
4. (abandon this change)

File: packages/aws/data_stream/billing/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -15,5 +15,11 @@ processors:
...
 on_failure:
   - set:
+      field: event.kind
+      value: pipeline_error
+  - append:
       field: error.message
-      value: "{{ _ingest.on_failure_message }}"
+      value: >-
+        Processor '{{{ _ingest.on_failure_processor_type }}}'
+        {{{#_ingest.on_failure_processor_tag}}}with tag '{{{ _ingest.on_failure_processor_tag }}}'
```
Comment (agithomas) [inline packages/aws/data_stream/billing/elasticsearch/ingest_pipeline/default.yml:24]: nit:

For a few values, ` with tag` (whitespace in front of `with`) is used, for a few this whitespace is missing.

Comment (efd6) [reply packages/aws/data_stream/billing/elasticsearch/ingest_pipeline/default.yml:24]: Thanks, I thought I got all of them with a s///g.

**Review by constanca-m — APPROVED** [review]

**Comment by efd6** [comment]
@ishleenk17 and @constanca-m it would be really helpful if you could also check out the blocking PR #11397 that this depends on.

**Pull request merged**